File: ruby/Strategy/csv_extraction.rb
require_relative 'extraction'

class CSVExtraction < Extraction

  def initialize
  end

  def extract
    open
    print "I will extract by seperating row by commas\n"
    close
  end

  private
  def open
    print "Opening CSV file\n"
  end

  private
  def close
    print "Closing CSV file\n"
  end
end

File: java/Strategy/src/CSVExtraction.java

class CSVExtraction implements Extraction {

	public void extract() {
		// complete the method below
		
		open();

		System.out.println("I will extract by separating row by commas.");

		close();
	}

	public void open() {

		System.out.println("Opening CSV file");
	}

	public void close() {

		System.out.println("Closing CSV file");
	}
}